src/analysis/mean_inequalities.lean

```diff
@@ -315,6 +315,50 @@ begin
     rpow_one, div_self hG_zero], },
 end
 
+/-- Hölder inequality: the scalar product of two functions is bounded by the product of their
+`L^p` and `L^q` norms when `p` and `q` are conjugate exponents. A version for `nnreal`-valued
+functions. For an alternative version, convenient if the infinite sums are already expressed as
+`p`-th powers, see `inner_le_Lp_mul_Lq_has_sum`. -/
+theorem inner_le_Lp_mul_Lq_tsum {f g : ι → ℝ≥0} {p q : ℝ} (hpq : p.is_conjugate_exponent q)
+  (hf : summable (λ i, (f i) ^ p)) (hg : summable (λ i, (g i) ^ q)) :
+  summable (λ i, f i * g i) ∧
+  ∑' i, f i * g i ≤ (∑' i, (f i) ^ p) ^ (1 / p) * (∑' i, (g i) ^ q) ^ (1 / q) :=
```

Comment:
It would be nice to have two more lemmas which each give one of the terms in this conjunction: one for summable, the other for tsum. Since the proof of both parts is the same, I don't suggest that you break this theorem in two, but rather that you keep this one and add two corollaries.

Same thing for the other `summable (...) ∧ ...` theorems

Reply:
Good idea, done!